static/src/javascripts/projects/commercial/modules/liveblog-adverts.ts

```diff
@@ -105,23 +112,23 @@ const fill = (rules) =>
 			if (el && el.previousSibling instanceof HTMLElement) {
 				firstSlot = el.previousSibling;
 			} else {
-				firstSlot = null;
+				firstSlot = undefined;
 			}
 			startListening();
 		} else {
-			firstSlot = null;
+			firstSlot = undefined;
 		}
 	});
 
 const onUpdate = () => {
 	stopListening();
-	Promise.resolve(getSpaceFillerRules(WINDOWHEIGHT, true)).then(fill);
+	void Promise.resolve(getSpaceFillerRules(WINDOWHEIGHT, true)).then(fill);
```

Comment:
I think this is just a lot of indirection, you can keep this all synchronous:

```suggestion
	const rules = getSpaceFillerRules(WINDOWHEIGHT, true);
	fill(rules);
```

or even the following if you like terse:

```suggestion
	fill(getSpaceFillerRules(WINDOWHEIGHT, true));
```

Reply:
Ah you're right this is pretty convoluted. 
It looks like since `onUpdate` is being passed to `addEventListener` it needs to return void though. How about this..

Reply:
Not returning, returning nothing and returning `undefined` are all equivalent to returning `void`.

Reply:
Sorry my comment on the return type of `onUpdate` wasn't super clear.
The problem with those two suggestions is that they don't handle the promise returned by `fill` which gives us this ts error
```
Promises must be handled appropriately or explicitly marked as ignored with the `void` operator.eslint[@typescript-eslint/no-floating-promises](https://github.com/typescript-eslint/typescript-eslint/blob/v4.33.0/packages/eslint-plugin/docs/rules/no-floating-promises.md)
```
 I think we either need to do `void` `fill` or `await` it which I think amount to the same thing given how we're using `onUpdate`.